modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -0,0 +1,135 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_VIDEO_HPP
+#define OPENCV_GAPI_VIDEO_HPP
+
+#include <utility> // std::tuple
+
+#include <opencv2/gapi/gkernel.hpp>
+
+
+/** \defgroup gapi_video G-API Video processing functionality
+ */
+
+namespace cv { namespace gapi {
+namespace  video
+{
+    using GOptFlowLKOutput = std::tuple<cv::GArray<cv::Point2f>,
+                                        cv::GArray<uchar>,
+                                        cv::GArray<float>>;
+
+    G_TYPED_KERNEL(GCalcOptFlowLK,
+                   <GOptFlowLKOutput(GMat,GMat,cv::GArray<cv::Point2f>,cv::GArray<cv::Point2f>,Size,
+                                     int,TermCriteria,int,double)>,
+                   "org.opencv.video.calcOpticalFlowPyrLK")
+    {
+        static std::tuple<GArrayDesc,GArrayDesc,GArrayDesc> outMeta(GMatDesc,GMatDesc,GArrayDesc,
+                                                                    GArrayDesc,const Size&,int,
+                                                                    const TermCriteria&,int,double)
+        {
+            return std::make_tuple(empty_array_desc(), empty_array_desc(), empty_array_desc());
+        }
+
+    };
+
+    G_TYPED_KERNEL(GCalcOptFlowLKForPyr,
+                   <GOptFlowLKOutput(cv::GArray<cv::GMat>,cv::GArray<cv::GMat>,
+                                     cv::GArray<cv::Point2f>,cv::GArray<cv::Point2f>,Size,int,
+                                     TermCriteria,int,double)>,
+                   "org.opencv.video.calcOpticalFlowPyrLKForPyr")
+    {
+        static std::tuple<GArrayDesc,GArrayDesc,GArrayDesc> outMeta(GArrayDesc,GArrayDesc,
+                                                                    GArrayDesc,GArrayDesc,
+                                                                    const Size&,int,
+                                                                    const TermCriteria&,int,double)
+        {
+            return std::make_tuple(empty_array_desc(), empty_array_desc(), empty_array_desc());
+        }
+    };
+} //namespace video
+
+//! @addtogroup gapi_video
+//! @{
+/** @brief Calculates an optical flow for a sparse feature set using the iterative Lucas-Kanade
+method with pyramids.
+
+See @cite Bouguet00 .
+
+@note Function textual ID is "org.opencv.video.calcOpticalFlowPyrLK"
+
+@param prevImg first 8-bit input image (GMat) or pyramid (GArray<GMat>) constructed by
+buildOpticalFlowPyramid.
```

Comment:
> or pyramid (GArray<GMat>) constructed by buildOpticalFlowPyramid

Which one? I don't see `buildOpticalFlowPyramid` in GAPI API.

Reply:
This kernel will be implemented later based on cv::buildOpticalFlowPyramid(); although, it works even now if you build pyramids with cv:: function and give them to this kernel (as it is implemented in tests)